presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java

```diff
@@ -105,15 +112,34 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
         return Result.ofPlanNode(
                 new AggregationNode(
                         context.getIdAllocator().getNextId(),
-                        new ProjectNode(
+                        new FilterNode(
```

Comment:
add `FilterNode` above `ProjectNode` using `maskSymbols`:
```
Expression predicate = TRUE_LITERAL;
if (!node.hasNonEmptyGroupingSet()) {
    predicate = combineDisjunctsWithDefault(maskSymbols.build(), TRUE_LITERAL));
}
new FilterNode(
  ...,
  predicate,
...
```

This way we won't do redundant computations of expressions

Reply:
This also needs to consider if any aggregate without a FILTER clause is present, made the change and fixed